### DISCUSSION TITLE
Css vriable in style prop

### DISCUSSION THREAD

**Discussion by gradinarot** [mantinedev/mantine#4607] [Q&A] [answered]
In v7 of Mantine I see a lot of css variable in the style prop:
```
<button style="--button-height:var(--button-height-sm);--button-padding-x:var(--button-padding-x-sm);--button-fz:var(--mantine-font-size-sm);--button-radius:var(--mantine-radius-default);--button-bg:var(--mantine-color-blue-filled);--button-hover:var(--mantine-color-blue-filled-hover);--button-color:var(--mantine-color-white);--button-bd:calc(0.0625rem * var(--mantine-scale)) solid transparent" class="mantine-focus-auto mantine-active m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root" data-variant="filled" data-size="sm" type="button"><span class="m-80f1301b mantine-Button-inner"><span class="m-811560b9 mantine-Button-label"></span></span></button>
```

Can we somehow remove this if we don't need theming?

**Comment by mattaningram** [comment]
I don't understand why variables are being declared on elements like this and not using a class? Every single instance of a component redeclares its defaults inline instead of using a class? That's a huge amount of unnecessary duplicative code.

**Reply by rtivital** [reply]
These variables are declared because they are configurable from props and it is not possible to know these values in advance.

**Reply by gradinarot** [reply]
can't we put them into a class?

**Reply by rtivital** [reply, marked as answer]
No, it is not possible
